controllers/etcdcluster_controller.go

```diff
@@ -719,6 +745,52 @@ func (r *EtcdClusterReconciler) reconcile(
 		return result, smCreatedEvent, nil
 	}
 
+	schedule, found := r.Schedules.Read(string(cluster.UID))
+	if !found {
+		id, err := r.CronHandler.AddFunc("*/1 * * * *", func() {
```

Comment:
Does it make sense to create a separate function for this? I think in a separated function with early returns this code should be much simpler.

Reply:
Done